File: ManiFest/ManiFest.Services/Database/Review.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ManiFest.Services.Database
{
    public class Review
    {
        [Key]
        public int Id { get; set; }

        [Required]
        [Range(1, 5)]
        public int Rating { get; set; }

        [MaxLength(1000)]
        public string? Comment { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Required]
        public int FestivalId { get; set; }
        [ForeignKey("FestivalId")]
        public Festival Festival { get; set; } = null!;

        [Required]
        public int UserId { get; set; }
        [ForeignKey("UserId")]
        public User User { get; set; } = null!;
    }
}

File: ManiFest/UI/manifest_desktop/lib/model/review.dart
import 'package:json_annotation/json_annotation.dart';

part 'review.g.dart';

@JsonSerializable()
class Review {
  final int id;
  final int rating;
  final String? comment;
  final DateTime createdAt;
  final int festivalId;
  final String festivalTitle;
  final int userId;
  final String username;
  final String userFullName;
  final String? festivalLogo;

  const Review({
    this.id = 0,
    this.rating = 0,
    this.comment,
    required this.createdAt,
    this.festivalId = 0,
    this.festivalTitle = '',
    this.userId = 0,
    this.username = '',
    this.userFullName = '',
    this.festivalLogo,
  });

  factory Review.fromJson(Map<String, dynamic> json) => _$ReviewFromJson(json);
  Map<String, dynamic> toJson() => _$ReviewToJson(this);
}
